CMakeLists.txt

```diff
@@ -1,6 +1,11 @@
-cmake_minimum_required(VERSION 3.12.0 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.11.0 FATAL_ERROR)
 
 option(IRODS_ENABLE_430_COMPATIBILITY "Enables compatibility with iRODS 4.3.0." YES) # Default to YES for community.
+option(IRODS_ENABLE_42X_COMPATIBILITY "Enables compatibility with iRODS 4.2.X." NO)  # Default to NO for community.
+
+if (IRODS_ENABLE_430_COMPATIBILITY AND IRODS_ENABLE_42X_COMPATIBILITY)
+  message(FATAL_ERROR "IRODS_ENABLE_430_COMPATIBILITY and IRODS_ENABLE_42X_COMPATIBILITY options are incompatible.")
+endif()
```

Comment:
If we're going to take the single-branch approach, instead of having the user specify which version of iRODS to support, I would recommend just checking the version numbers in the irods cmake config, which is imported by find_package.